### DISCUSSION TITLE
Operation returned an invalid status code 'Unauthorized' while trying to do json patch for custom resource objects

### DISCUSSION THREAD

**Discussion by srinivasmummareddy** [kubernetes-client/csharp#1025] [General]
The following is the stack trace of exception

"Error","@x":"k8s.Autorest.HttpOperationException: Operation returned an invalid status code 'Unauthorized'\n at k8s.Kubernetes.SendRequestRaw(String requestContent, HttpRequestMessage httpRequest, CancellationToken cancellationToken)\n at k8s.AbstractKubernetes.k8s.ICustomObjectsOperations.PatchNamespacedCustomObjectWithHttpMessagesAsync(Object body, String group, String version, String namespaceParameter, String plural, String name, String dryRun, String fieldManager, Nullable1 force, IReadOnlyDictionary2 customHeaders, CancellationToken cancellationToken)\n at k8s.CustomObjectsOperationsExtensions.PatchNamespacedCustomObjectAsync(ICustomObjectsOperations operations, Object body, String group, String version, String namespaceParameter, String plural, String name, String dryRun, String fieldManager, Nullable`1 force, CancellationToken cancellationToken)

Note: The particular pod has a service account which has permission to patch the crd.

Not able to figure out the exact reason

I'm using Kubernetes client : 8.0.68

**Comment by tg123** [comment]
could you please test with kubectl auth can-i?
